libs/ledger-live-common/src/families/vechain/utils/transaction-utils.ts

```diff
@@ -0,0 +1,107 @@
+import { getEnv } from "../../../env";
+import network from "../../../network";
+import { HEX_PREFIX, ZERO_ADDRESS } from "../constants";
+import crypto from "crypto";
+import BigNumber from "bignumber.js";
+import { Transaction as ThorTransaction } from "thor-devkit";
+import params from "../contracts/abis/params";
+import {
+  BASE_GAS_PRICE_KEY,
+  PARAMS_ADDRESS,
+  VTHO_ADDRESS,
+} from "../contracts/constants";
+import { Query } from "../api/types";
+import { query } from "../api/sdk";
+import { isValid } from "./address-utils";
+import { Transaction } from "../types";
+
+const BASE_URL = getEnv("API_VECHAIN_THOREST");
+
+/**
+ * Get the block ref to use in a transaction
+ * @returns the block ref of head
+ */
+export const getBlockRef = async (): Promise<string> => {
+  const { data } = await network({
+    method: "GET",
+    url: `${BASE_URL}/blocks/best`,
+  });
+
+  return data.id.slice(0, 18);
+};
+
+/**
+ * Generate a Unique ID to be used as a nonce
+ * @returns a unique string
+ */
+export const generateNonce = (): string => {
+  const randBuffer = crypto.randomBytes(Math.ceil(4));
+  if (!randBuffer) throw Error("Failed to generate random hex");
+  return `${HEX_PREFIX}${randBuffer.toString("hex").substring(0, 8)}`;
+};
+
+/**
+ * Estimate the gas that will be used by the transaction.
+ * TODO: Investigate if this implementation is sufficient.
```

Comment:
I think we can remove this TODO and the comment below it. It was a reference to the previous more naive implementation